fractal_server/app/routes/auth.py

```diff
@@ -53,10 +56,25 @@
 )
 
 
-@router_auth.get("/whoami", response_model=UserRead)
-async def whoami(
-    user: User = Depends(current_active_user),
+@router_auth.patch("/me", response_model=UserRead)
+async def patch_current_user(
+    user_update: UserUpdate,
+    current_user: User = Depends(current_active_user),
+    user_manager=Depends(get_user_manager),
 ):
+    updates = {
+        k: getattr(user_update, k)
```

Comment:
# Comment 1

As of recent PRs, I suggest we try to _not_ be smart, and proceed with the simplest possible code.
Here we are using 5 lines, a for loop, a `getattr` function, a second (identical) `getattr` function, the construction of a list object, and a conditional.
An alternative option would be
```python
updates = {}
if user_update.cache_dir is not None:
    updates["cache_dir"] =  user_update.cache_dir
if user_update.password is not None:
    updates["password"] =  user_update.password
```
or am I missing something?

# Comment 2
The current solution does not raise any error if the client tries to modify other attributes. I propose that we define a more explicit Pydantic schema (e.g. `UserUpdateStrict`) which only includes the attributes that can be modified.
I propose that we are even more strict